src/crimson/os/seastore/collection_manager/collection_flat_node.h

```diff
@@ -153,10 +153,11 @@ struct CollectionNode
   }
 
   ceph::bufferlist get_delta() final {
-    assert(!delta_buffer.empty());
```

Comment:
The assert was here because I suppose the delta should be aways valid if the extent has been mutated.

In another word, I thought that the extent should not be mutated/duplicated if it doesn't need to be changed, otherwise it implies a waste of effort.

Or probably I'm missing something -- why is this possible to this type of node?

Reply:
> The assert was here because I suppose the delta should be aways valid if the extent has been mutated.
> 
> In another word, I thought that the extent should not be mutated/duplicated if it doesn't need to be changed, otherwise it implies a waste of effort.

This is exactly what the problem is. In the collection node manager, collection nodes are first mutated and then checked again whether it would overflow, and if it does overflow, the mutation_pending node is untouched and a new node with twice the size is created. Then we get a mutation_pending node that doesn't actually have any deltas.

> 
> Or probably I'm missing something -- why is this possible to this type of node?

Reply:
OK, so this change should be good as a temporary workaround.

But I think a `// FIXME` discription is needed in the code.